requirements/requirements-dev.txt

```diff
@@ -28,4 +28,4 @@ bqskit==1.1.1
 seaborn==0.13.0
 stim==1.13.0
 stimcirq==1.13.0
-pyqrack==1.30.30
\ No newline at end of file
+pyqrack==1.31.1
```

Comment:
@natestemen we should make this dependency less strict with `~=`

Reply:
We've generally tried to keep dependencies that are required for documentation to be pinned exactly for reproducibility, and so CI/RTD doesn't fail randomly when a new version is released that doesn't adhere to SemVer.

Reply:
but this is a UF-owned package :) if it doesn't adhere to SemVer, we know whom to bug.

Reply:
Not a hill I'm willing to die on, but I feel like skipping patch/minor dependencies solves the issue of there being too many dependabot alerts.

I like pinning exactly for docs for reproducibility sake.